### PR TITLE
Maven plugin 🎉

### PR DESCRIPTION
**Pull request by patxibocos**

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -15,23 +15,38 @@ jobs:
...
-      run: echo ::set-env name=LOCAL_MAVEN_URL::$(echo $GITHUB_WORKSPACE/repository)
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
 
     - name: Build and publish generator in local Maven repository
-      run: ./generator/gradlew -p=generator publish
+      run: ./gradlew publish
+      working-directory: ./generator
```
Comment (ralphpina) [inline .github/workflows/build_test.yml:25]: This is a nice API 👍.

File: generator/build.gradle.kts [hunk 2]
```diff
@@ -61,8 +61,6 @@ publishing {
         }
     }
     repositories {
-        maven {
-            url = uri(System.getenv("LOCAL_MAVEN_URL"))
-        }
+        mavenLocal()
```
Comment (ralphpina) [inline generator/build.gradle.kts:64]: Nicer :)

File: maven-plugin/src/main/kotlin/com/patxi/poetimizely/maven/plugin/GeneratorMojo.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import com.patxi.poetimizely.generator.codeForFeatures
+import java.io.File
+import kotlinx.coroutines.runBlocking
+import org.apache.maven.plugin.AbstractMojo
+import org.apache.maven.plugins.annotations.Mojo
+import org.apache.maven.plugins.annotations.Parameter
+import org.apache.maven.project.MavenProject
+
+@Mojo(name = "poetimize")
+class GeneratorMojo : AbstractMojo() {
```
Comment (ralphpina) [inline maven-plugin/src/main/kotlin/com/patxi/poetimizely/maven/plugin/GeneratorMojo.kt:13]: Man... it's so tempting to try to write a superclass for this and [GeneratorTask.kt](https://github.com/patxibocos/poetimizely/pull/41/files#diff-dedb216aaa13107b11af818807e0f44a), but so wrong.

Comment (patxibocos) [reply maven-plugin/src/main/kotlin/com/patxi/poetimizely/maven/plugin/GeneratorMojo.kt:13]: Exactly, I was also considering on moving the logic of writing the code in a file into the generator module, but we can also do that later :)

**Comment by patxibocos** [comment]
> Never saw a maven plugin implementation until now 👍.
> 
> This is so fucking easy because of how well designed this is 🙌.

🙂 What I have learned building a Maven plugin is how much I prefer using Gradle + Kotlin DSL versus touching the pom.xml 😂

**Pull request merged**